servicetalk-http-api/src/test/java/io/servicetalk/http/api/AbstractHttpRequestMetaDataTest.java

```diff
@@ -801,6 +801,52 @@ void testEncodeToRequestTargetWithAbsoluteFormWithUserInfo() {
         assertEquals("foo=new&abc=def", fixture.query()); // check cached value
     }
 
+    @Test
+    void testGetFragment() {
+        createFixture("/some/path#fragment");
+        assertEquals("fragment", fixture.fragment());
+    }
+
+    @Test
+    void testSetFragment() {
+        createFixture("/some/path");
+        fixture.fragment("fragment");
+        assertEquals("fragment", fixture.fragment());
+        assertEquals("/some/path#fragment", fixture.requestTarget());
+    }
+
+    @Test
+    void testSetFragmentWithQuery() {
+        createFixture("/some/path?query=value");
+        fixture.fragment("fragment");
+        assertEquals("fragment", fixture.fragment());
+        assertEquals("/some/path?query=value#fragment", fixture.requestTarget());
+    }
+
+    @Test
+    void testSetFragmentAbsoluteForm() {
+        createFixture("http://my.site.com/some/path?foo=bar&abc=def&foo=baz");
+        fixture.fragment("fragment");
+        assertEquals("fragment", fixture.fragment());
+        assertEquals("http://my.site.com/some/path?foo=bar&abc=def&foo=baz#fragment", fixture.requestTarget());
+    }
+
+    @Test
+    void testOverwriteFragment() {
+        createFixture("/some/path?query=value#fragment");
+        fixture.fragment("overwrite");
+        assertEquals("overwrite", fixture.fragment());
+        assertEquals("/some/path?query=value#overwrite", fixture.requestTarget());
+    }
+
+    @Test
+    void testClearFragment() {
+        createFixture("/some/path?query=value#fragment");
+        fixture.fragment(null);
```

Comment:
Thanks for the good test coverage! Consider 3 more tests:
1. When requestTarget has an empty fragment: `/some/path#`, `assertThat(fixture.fragment(), is(emptyString()))`
2. When users set an empty string as a fragment
3. When users set the same fragment